source/libtomo/accel/cxx/mlem.cc

```diff
@@ -94,13 +94,17 @@ cxx_mlem(const float* data, int dy, int dt, int dx, const float* center,
     {
         if(opts.device.key == "gpu")
         {
+#if defined(TOMOPY_USE_CUDA)
             mlem_cuda(data, dy, dt, dx, center, theta, recon, ngridx, ngridy, num_iter,
                       &opts);
+#endif
         }
         else
         {
+#if defined(TOMOPY_USE_OPENCV)
             mlem_cpu(data, dy, dt, dx, center, theta, recon, ngridx, ngridy, num_iter,
                      &opts);
+#endif
```

Comment:
We should throw an exception or abort here instead of doing nothing to prevent segmentation faults.

Reply:
Device key can no longer be 'gpu' or 'cpu' if the supporting library wasn't included.